src/components/StakingVaults/AllEarnOpportunities.tsx

```diff
@@ -42,14 +41,15 @@ export const AllEarnOpportunities = () => {
 
   const handleClick = useCallback(
     (opportunity: EarnOpportunityType) => {
-      const { type, provider, contractAddress, chain, tokenAddress, rewardAddress, assetId } =
+      const { type, provider, contractAddress, chainId, tokenAddress, rewardAddress, assetId } =
         opportunity
       if (!isConnected && walletInfo?.deviceId !== DemoConfig.name) {
         dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
         return
       }
 
-      if (chain === ChainTypes.Cosmos) {
+      // TODO(gomes): when we find new home for "supported chainIds", use it here. Don't metastize supportedChainId imports from caip
+      if (chainId === 'cosmos:cosmoshub-4') {
```

Comment:
What about the chainId constants exported from caip?

```suggestion
      if (chainId === cosmosChainId) {
```